### PR TITLE
Refactor output for removed `addon_profile` block. Deprecate legacy dot index syntax in outputs.

### LINKED ISSUE #180: `addon_profile` has been removed from provider 3.x so we cannot export it in the outputs anymore.
As @zioproto pointed out in #179, the `addon_profile` has been removed in this version of the provider.

We need to update the outputs so the user can access these data, meanwhile minimize the effort for the users who're using the output right now.

### PR DESCRIPTION
**Pull request by lonegunmanb**
This patch will fix both #180 and #182.

As `addon_profile` was added by @davidkarlsen, would you please give this pr a review to see whether the new output object can work with your code? Thanks!

### CODE REVIEW

File: outputs.tf [hunk 1]
```diff
@@ -64,29 +63,41 @@ output "kubelet_identity" {
...
-  value = length(azurerm_kubernetes_cluster.main.kube_admin_config) > 0 ? azurerm_kubernetes_cluster.main.kube_admin_config.0.username : ""
+  value = try(azurerm_kubernetes_cluster.main.kube_admin_config[0].username, "")
 }
 
 output "admin_password" {
-  value = length(azurerm_kubernetes_cluster.main.kube_admin_config) > 0 ? azurerm_kubernetes_cluster.main.kube_admin_config.0.password : ""
+  value = try(azurerm_kubernetes_cluster.main.kube_admin_config[0].password, "")
 }
 
 output "addon_profile" {
```
[the remark this answers is not in the document]
Comment (zioproto) [reply outputs.tf:89]: I would drop the `addon_profile` string in the outputs and add information about this change in the Changelog file

[the remark this answers is not in the document]
Comment (davidkarlsen) [reply outputs.tf:89]: yes, `addon_profile` makes no sense with the updated provider. I added the export in order to get hold of the identity of the AGIC, in order to be able to do role assignments:
```hcl
principal_id         = module.aks.addon_profile.ingress_application_gateway[0].ingress_application_gateway_identity[0].object_id
  scope                = module.aks.addon_profile.ingress_application_gateway[0].gateway_id
```

as long as we still get hold of those values anything is fine.

[the remark this answers is not in the document]
Comment (lonegunmanb) [reply outputs.tf:89]: Agreed!

**Comment by zioproto** [comment]
LGTM

**Pull request merged**